### ISSUE TITLE
Add function for resizing to frame without cropping.

### ISSUE THREAD

**Issue by c-edw** [PistonDevelopers/image#769]
If I resize an image to fit a frame using `resize_to_fill`, the image is cropped if the image is not a perfect square. Can we implement a function for resizing to fill a frame without cropping?

**Comment by c-edw** [comment]
The `resize` function will make the image fit the bounds, but not fill it. Let's say I have a 64x32 image which I resize to 128x128. The output image will be 128x64. I need an output of 128x128, with the image in the center of the frame.

`resize_exact` resizes the image to fill a frame but crops out parts outside the frame. Using this, I will get my 128x128 image, but there will be 32 pixels on the top and bottom of the image cut out.

**Comment by HeroicKatora** [comment]
You can `resize`, then write the result in a `SubImage` of the correct size. That makes it also possible to choose an arbitrary background.

**Comment by c-edw** [comment]
@HeroicKatora Yeah, I see that it would bloat the crate a bit. Could we compromise and have an operation for overlaying images in a centered matter?

**Comment by lukehsiao** [comment]
I can understand not wanting to bloat the API. That said, an example of how to achieve this operation using the existing API would be helpful.

**Comment by fintelia** [comment]
@lukehsiao blitting operations tend not to be that hard to implement with simple loops, but there are a bunch of variations that people can want. In the example described above, I'd implement it something like:

```rust
fn copy_onto(image: &RgbaImage, background: RgbaImage) -> RgbaImage {
    let resized = imageops::resize(&image, 128, 64, filter);
    
    for x in 0..128 {
        for y in 0..64 {
            background.put_pixel(x, y+32, resized.get_pixel(x,y));
        }
    } 

    background
}
```

**Comment by c-edw** [comment]
Probably not too good of an idea to loop through individual pixels like that. Here's the implemention I wrote. You'll need to tweak it a bit since I ripped out straight from my project:

```rust
let width = original.width() as f32;
let height = original.height() as f32;

// Find the optimum output size.
let size = (width.max(height)).min(MAX_OUTPUT_SIZE);
let scalar = (size / width).min(size / height);

let (xscaled, yscaled) = (scalar * width, scalar * height);
let (xoverlay, yoverlay) = ((size - xscaled) / 2.0, (size - yscaled) / 2.0);

let resized = imageops::resize(
    &original,
    xscaled as u32,
    yscaled as u32,
    FilterType::Lanczos3,
);

let mut rescaled = ImageBuffer::new(size as u32, size as u32) as RgbImage;
imageops::overlay(&mut rescaled, &resized, xoverlay as u32, yoverlay as u32);
```

**Comment by fintelia** [comment]
What's wrong with looping through individual pixels? [That's literally what imageops::overlay does.](https://docs.rs/image/0.21.0/src/image/imageops/mod.rs.html#127)

**Comment by c-edw** [comment]
I would've expected this function just to be a memcpy. I think at the very least rayon should be used to parallelise those loops.
